doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+    "\n",
+    "- **Concept drift**: This occurs when the process generating $y$ from $x$ has changed, such that $P(\\mathbf{Y}|\\mathbf{X}) \\ne P_{ref}(\\mathbf{Y}|\\mathbf{X})$. An example of this could be if a change in the wider economic climate resulted in a change in the risk for a given mortgage application. It is possible that the model might no longer give a suitable approximation of the true process. \n",
+    "\n",
+    "Below, these different types of drift are visualised for a simple two-dimensional classification problem. It is possible for a drift to fall under more than one category, for example the *prior drift* below also happens to be a case of *covariate drift*. \n",
+    "\n",
+    "![2d drift example](images/bg_2d_drift.png) \n",
+    "\n",
+    "It is relatively easy to spot drift by eyeballing these figures here. However, the task becomes considerably harder for high-dimensional real problems, especially since real-time ground truths are not typically available. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Detecting drift <a class=\"anchor\" id=\"drift_detection\"></a>\n",
+    "\n",
+    "[Alibi Detect](https://github.com/SeldonIO/alibi-detect) offers a wide array of methods for detecting drift (see [here](../overview/algorithms.md)), many of which are examined in the NeurIPS 2019 paper [Failing Loudly: An Empirical Study of Methods for Detecting Dataset Shift](https://arxiv.org/abs/1810.11953). Generally, these aim to determine whether the distribution $P(\\mathbf{x})$ has drifted from a reference distribution $P_{ref}(\\mathbf{x})$, where $\\mathbf{x}$ may represent input data $\\mathbf{X}$, true output data $\\mathbf{Y}$, or some form of model output, depending on what type of drift we wish to detect. \n",
+    "\n",
+    "Due to natural randomness in the process being modelled, we don't necessarily expect observations $\\mathbf{x}_1,\\dots,\\mathbf{x}_N$ drawn from $P(\\mathbf{x})$ to be identical to $\\mathbf{x}^{ref}_1,\\dots,\\mathbf{x}^{ref}_M$ drawn from $P_{ref}(\\mathbf{x})$. To decide whether differences between $P(\\mathbf{x})$ and $P_{ref}(\\mathbf{x})$ are due to drift or just natural randomness in the data, *statistical two-sample hypothesis* testing is used, with the null hypothesis $P(\\mathbf{x})=P_{ref}(\\mathbf{x})$. If the p-value obtained is below a given threshold, the null is rejected and the alternative hypothesis $P(\\mathbf{x}) \\ne P_{ref}(\\mathbf{x})$ is accepted, suggesting drift is occuring. \n",
+    "\n",
+    "Since $\\mathbf{x}$ is often high-dimensional (even a 200 x 200 greyscale image has 40k dimensions!), performing hypothesis testing in the full-space is often computationally intractable. Instead, the pipeline below is often used, with dimension reduction as a pre-processing step.  \n",
```

Comment:
Intractable or unsuitable for the statistical test which is performed. E.g. a K-S test can pretty quickly be computed on relatively high dimensions, but this will not result in a great test because it just compares a lot of individual stats on the marginals and average them somehow.